muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/coursepicker/CoursePickerRESTService.java

```diff
@@ -581,7 +617,28 @@ private boolean getCanSignup(WorkspaceEntity workspaceEntity) {
     }
   }
   
-  private CoursePickerWorkspace createRestModel(WorkspaceEntity workspaceEntity, String name, String nameExtension, String description, String educationTypeName, boolean canSignup, boolean isCourseMember) {
+  private boolean getIsAlreadyEvaluated(WorkspaceEntity workspaceEntity) {
+    if (sessionController.isLoggedIn()) {
+      WorkspaceUserEntity workspaceUserEntity = workspaceUserEntityController.findWorkspaceUserByWorkspaceEntityAndUserIdentifier(workspaceEntity, sessionController.getLoggedUserEntity().defaultSchoolDataIdentifier());
+      Boolean isUnassessed = false;
+      if (workspaceUserEntity != null) {
+        WorkspaceRoleEntity workspaceRoleEntity = workspaceUserEntity.getWorkspaceUserRole();
+        WorkspaceRoleArchetype archetype = workspaceRoleEntity.getArchetype();
+        if (archetype.equals(WorkspaceRoleArchetype.STUDENT)) {
+          WorkspaceAssessmentState assessmentState = assessmentRequestController.getWorkspaceAssessmentState(workspaceUserEntity);
+          if (!assessmentState.getState().equals(WorkspaceAssessmentState.UNASSESSED)) {
+            isUnassessed = true;
+          }
+        }  
+      }
+      return isUnassessed;
+    }
+    else {
+      return false;
+    }
+  }
```

Comment:
The rest of this block irrelevant if you have `boolean isEvaluated = false;` as the very first line of the method. Just return that variable as the last line of the method. The `if (assessmentState().getState() == WorkspaceAssessmentState.PASS)` block flips that initial presumption to `true`, if need be.